lib/core/test.js

```diff
@@ -48,11 +48,20 @@ export class Test {
     this.finishWithPendingStatus();
   }
 
+  skip() {
+    this.markExplicitlySkipped(TestResult.explicitlySkip());
+  }
+
   markSkipped(skippedResult) {
     this.setResult(skippedResult);
     this.#callbacks.whenSkipped(this);
   }
 
+  markExplicitlySkipped(explicitlySkippedResult) {
+    this.setResult(explicitlySkippedResult);
+    this.#callbacks.whenSkipped(this);
```

Comment:
we don't have to run the callback here, because it can be run in `finishWithSkippedStatus` after evaluating the result. This behavior ends up calling the callback twice, see my review summary for a test that reproduces the issue